src/main/java/rx/internal/util/SubscriptionList.java

```diff
@@ -106,20 +106,7 @@ private static void unsubscribeFromAll(Collection<Subscription> subscriptions) {
                 es.add(e);
             }
         }
-        if (es != null) {
-            if (es.size() == 1) {
-                Throwable t = es.get(0);
-                if (t instanceof RuntimeException) {
-                    throw (RuntimeException) t;
-                } else {
-                    throw new CompositeException(
-                            "Failed to unsubscribe to 1 or more subscriptions.", es);
```

Comment:
This looks like a mistake to have been a `CompositeException` since it's not composite, there is only 1. If it was a `RuntimeException` on the other side of the branch we just threw it, so we should just be using `propagate` here. 

Perhaps the reason this was done was to add a better error message? That's something an overloaded `propagate` could add.